lib/sda/tree.go

```diff
@@ -285,12 +286,22 @@ func (il *EntityList) GenerateBigNaryTree(N, nodes int) *Tree {
 				children = N
 			}
 			parent.Children = make([]*TreeNode, children)
+			parentHost, _, _ := net.SplitHostPort(parent.Entity.Addresses[0])
 			for n := 0; n < children; n++ {
-				// TODO: this should check on the host-address
-				// of the Entity and not only the Id.
-				for il.List[elIndex].Id == parent.Entity.Id &&
+				// Check on host-address, so that no child is
+				// on the same host as the parent.
+				childHost, _, _ := net.SplitHostPort(il.List[elIndex].Addresses[0])
+				elIndexFirst := elIndex
+				for childHost == parentHost &&
 					ilLen > 1 {
 					elIndex = (elIndex + 1) % ilLen
+					// If we tried all hosts, it means we're using
+					// just one hostname, as we didn't find any
+					// other name
+					if elIndex == elIndexFirst {
+						break
+					}
+					childHost, _, _ = net.SplitHostPort(il.List[elIndex].Addresses[0])
```

Comment:
make this a local var `net.SplitHostPort(il.List[elIndex].Addresses[0])` (used several times), why is `n` never used in this loop?